apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/ValueResolver.kt

```diff
@@ -2,6 +2,6 @@ package com.apollographql.apollo.api.internal
 
 import com.apollographql.apollo.api.ResponseField
 
-interface FieldValueResolver<R> {
-  fun <T> valueFor(recordSet: R, field: ResponseField): T?
+interface ValueResolver<in R> {
```

Comment:
just curious why we renamed it? should it be more explicit `ResponseFieldValueResolver`?

Reply:
I think the rationale was something like:
- this is a small, narrow-scope class (there are 2 usages) so there's no need for a very long and descriptive type 
- a `Map` knows about `values`, not `ResponseFields`. So this class retrieves a "value" using the given "ResponseField". And not: this class retrieves a "ResponseFieldValue"

But TBH, given that it's a small API, I don't mind that much. We should rename `recordSet` to `map` though...